Description: Remove static feautures and patch build.rs
 This patch removes all unnecessary features and patches build.rs to only check for a system-provided liblcms2.
Author: Matthias Geiger <matthias.geiger1024@tutanota.de>
Forwarded: not-needed
Last-Update: 2023-07-03
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -55,27 +55,11 @@ doctest = false
 [dependencies.libc]
 version = "0.2.129"
 
-[build-dependencies.cc]
-version = "1.0.73"
-optional = true
-
 [build-dependencies.dunce]
 version = "1.0.2"
 
 [build-dependencies.pkg-config]
 version = "0.3.25"
-optional = true
-
-[features]
-default = [
-    "dynamic",
-    "static-fallback",
-    "parallel",
-]
-dynamic = ["pkg-config"]
-parallel = ["cc/parallel"]
-static = ["cc"]
-static-fallback = ["cc"]
 
 [badges.maintenance]
 status = "actively-developed"
--- a/src/build.rs
+++ b/src/build.rs
@@ -1,88 +1,7 @@
-use std::env;
-use std::env::consts;
-use std::path::Path;
+extern crate pkg_config;
 
-fn main() {
-    if let Ok(include_dir) = env::var("LCMS2_INCLUDE_DIR") {
-        println!("cargo:include={}", include_dir);
-    }
-
-    if let Some(lib_dir) = env::var_os("LCMS2_LIB_DIR") {
-        let lib_dir = Path::new(&lib_dir);
-        let dylib_name = format!("{}lcms2{}", consts::DLL_PREFIX, consts::DLL_SUFFIX);
-        if lib_dir.join(dylib_name).exists() ||
-           lib_dir.join("liblcms2.a").exists() ||
-           lib_dir.join("lcms2.lib").exists() {
-            println!("cargo:rustc-link-search=native={}", lib_dir.display());
-            println!("cargo:rustc-link-lib=lcms2");
-            return;
-        }
-    }
-
-    let requires_static_only = cfg!(feature = "static") || env::var("LCMS2_STATIC").is_ok();
-    if requires_static_only || (!configure_pkg_config() && cfg!(feature = "static-fallback")) {
-        compile_static();
-    }
-}
-
-#[cfg(feature = "dynamic")]
-fn configure_pkg_config() -> bool {
-    match pkg_config::probe_library("lcms2") {
-        Ok(info) => {
-            for path in info.include_paths {
-                println!("cargo:include={}", path.display());
-            }
-            true
-        },
-        Err(err) => {
-            println!("cargo:warning=pkg_config failed ({}). Falling back to static build.", err);
-            false
-        },
-    }
-}
-
-#[cfg(not(feature = "dynamic"))]
-fn configure_pkg_config() -> bool {
-    false
-}
-
-#[cfg(not(any(feature = "static", feature = "static-fallback")))]
-fn compile_static() {
-    println!("cargo:warning='static' feature of lcms2-sys is disabled, so the library won't be built, and probably won't work at all");
-    println!("cargo:rustc-link-lib=lcms2");
+fn main()
+{
+	pkg_config::probe_library("lcms2").unwrap();
 }
 
-#[cfg(any(feature = "static", feature = "static-fallback"))]
-fn compile_static() {
-    cc::Build::new()
-        .include("vendor/include")
-        .file("vendor/src/cmsalpha.c")
-        .file("vendor/src/cmscam02.c")
-        .file("vendor/src/cmscgats.c")
-        .file("vendor/src/cmscnvrt.c")
-        .file("vendor/src/cmserr.c")
-        .file("vendor/src/cmsgamma.c")
-        .file("vendor/src/cmsgmt.c")
-        .file("vendor/src/cmshalf.c")
-        .file("vendor/src/cmsintrp.c")
-        .file("vendor/src/cmsio0.c")
-        .file("vendor/src/cmsio1.c")
-        .file("vendor/src/cmslut.c")
-        .file("vendor/src/cmsmd5.c")
-        .file("vendor/src/cmsmtrx.c")
-        .file("vendor/src/cmsnamed.c")
-        .file("vendor/src/cmsopt.c")
-        .file("vendor/src/cmspack.c")
-        .file("vendor/src/cmspcs.c")
-        .file("vendor/src/cmsplugin.c")
-        .file("vendor/src/cmsps2.c")
-        .file("vendor/src/cmssamp.c")
-        .file("vendor/src/cmssm.c")
-        .file("vendor/src/cmstypes.c")
-        .file("vendor/src/cmsvirt.c")
-        .file("vendor/src/cmswtpnt.c")
-        .file("vendor/src/cmsxform.c")
-        .compile("liblcms2.a");
-
-    println!("cargo:include={}", dunce::canonicalize("vendor/include").unwrap().display());
-}
